images/win/scripts/Installers/Install-Haskell.ps1

```diff
@@ -16,32 +16,25 @@ Set-SystemVariable "CABAL_DIR" $cabalDir
 Add-MachinePathItem "$ghcupPrefix\ghcup\bin"
 Add-MachinePathItem "$cabalDir\bin"
 
-# Get 3 latest versions of GHC, use OData query for that since choco search has issues https://github.com/chocolatey/choco/issues/2271
-$ODataQuery = '$filter=(Title eq ''ghc'') and (IsPrerelease eq false)&$orderby=Version desc'
-$Url = "https://community.chocolatey.org/api/v2/Packages()?$ODataQuery"
-$ChocoVersionsOutput = [Version[]](Invoke-RestMethod -Uri $Url).properties.Version
-$LatestMajorMinor = $ChocoVersionsOutput | Group-Object { $_.ToString(2) } | Sort-Object { [Version]$_.Name } | Select-Object -last 3
-$VersionsList = $LatestMajorMinor | ForEach-Object { $_.Group | Select-Object -First 1 } | Sort-Object
+# Get 3 latest versions of GHC
+$Versions = ghcup list -t ghc -r | Where-Object {$_ -notlike "prerelease"}
+$VersionsOutput = [Version[]]($Versions | %{ $_.Split(' ')[1]; })
+$LatestMajorMinor = $VersionsOutput | Group-Object { $_.ToString(2) } | Sort-Object { [Version]$_.Name } | Select-Object -last 3
+$VersionsList = $LatestMajorMinor | ForEach-Object { $_.Group | Select-Object -Last 1 } | Sort-Object
 
 # The latest version will be installed as a default
 ForEach ($version in $VersionsList)
 {
     Write-Host "Installing ghc $version..."
     ghcup install ghc $version
-    ghcup set ghc $version
 }
 
-# Add default version of GHC to path, because choco formula updates path on user level
+# Add default version of GHC to path
 $DefaultGhcVersion = $VersionsList | Select-Object -Last 1
 $DefaultGhcShortVersion = ([version]$DefaultGhcVersion).ToString(3)
-$DefaultGhcPath = Join-Path $env:ChocolateyInstall "lib\ghc.$DefaultGhcVersion\tools\ghc-$DefaultGhcShortVersion\bin"
-# Starting from version 9 haskell installation directory is $env:ChocolateyToolsLocation instead of $env:ChocolateyInstall\lib
-if ($DefaultGhcShortVersion -notmatch '^[0-8]\.\d+.*')
-{
-    $DefaultGhcPath = Join-Path $env:ChocolateyToolsLocation "ghc-$DefaultGhcShortVersion\bin"
-}
-
+$DefaultGhcPath = Join-Path $env:GHCUP_INSTALL_BASE_PREFIX "ghcup\ghc\$DefaultGhcShortVersion\bin"
```

Comment:
this is not really needed... all binaries are in `Join-Path $env:GHCUP_INSTALL_BASE_PREFIX "ghcup\bin"` already, which we expose to path anyway. So I think we can remove this and the line below.

Reply:
updated. please check once more. thank you!

Reply:
one more thing, i think that '$cabalDir = "C:\cabal"' is now also false

Reply:
No... `$cabalDir` is where the cabal binary that resides in `/c/ghcup/bin/cabal` will write its runtime files into.